core/src/main/scala/caliban/introspection/Introspector.scala

```diff
@@ -67,14 +78,17 @@ object Introspector extends IntrospectionDerivation {
       .values
       .toList
       .sortBy(_.name.getOrElse(""))
+
+    val hasOneOf = types.exists(_._isOneOfInput)
+
     val resolver = __Introspection(
       __Schema(
         rootType.description,
         rootType.queryType,
         rootType.mutationType,
         rootType.subscriptionType,
         types,
-        directives ++ rootType.additionalDirectives
+        directives ++ (if (hasOneOf) List(oneOfDirective) else Nil) ++ rootType.additionalDirectives
```

Comment:
We can enable this by default. I thought we might want to add it only when defined in the schema at first

Reply:
I agree